google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/git/UploadSourceAction.java

```diff
@@ -467,8 +469,12 @@ public void run() {
         return false;
       }
 
-      Collection<VirtualFile> files2add = ContainerUtil.intersection(untrackedFiles, files2commit);
-      Collection<VirtualFile> files2rm = ContainerUtil.subtract(trackedFiles, files2commit);
+      Set<VirtualFile> files2CommitAsSet = new HashSet<>(files2commit);
+      Set<VirtualFile> untrackedFilesAsSet = new HashSet<>(untrackedFiles);
+      Set<VirtualFile> trackedFilesAsSet = new HashSet<>(trackedFiles);
+
+      Collection<VirtualFile> files2add = Sets.intersection(untrackedFilesAsSet, files2CommitAsSet);
```

Comment:
the use of '2' here kind of bugs me, but I guess thats not what this PR is about